// @pinecone-database/pinecone/dist/errors/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PineconeBatchUpsertError = exports.handleApiError = exports.extractMessage = exports.PineconeArgumentError = exports.PineconeConnectionError = exports.PineconeUnknownRequestFailure = exports.PineconeEnvironmentVarsNotSupportedError = exports.PineconeUnexpectedResponseError = exports.PineconeConfigurationError = void 0;
var config_1 = require("./config");
Object.defineProperty(exports, "PineconeConfigurationError", { enumerable: true, get: function () { return config_1.PineconeConfigurationError; } });
Object.defineProperty(exports, "PineconeUnexpectedResponseError", { enumerable: true, get: function () { return config_1.PineconeUnexpectedResponseError; } });
Object.defineProperty(exports, "PineconeEnvironmentVarsNotSupportedError", { enumerable: true, get: function () { return config_1.PineconeEnvironmentVarsNotSupportedError; } });
Object.defineProperty(exports, "PineconeUnknownRequestFailure", { enumerable: true, get: function () { return config_1.PineconeUnknownRequestFailure; } });
__exportStar(require("./http"), exports);
var request_1 = require("./request");
Object.defineProperty(exports, "PineconeConnectionError", { enumerable: true, get: function () { return request_1.PineconeConnectionError; } });
var validation_1 = require("./validation");
Object.defineProperty(exports, "PineconeArgumentError", { enumerable: true, get: function () { return validation_1.PineconeArgumentError; } });
var utils_1 = require("./utils");
Object.defineProperty(exports, "extractMessage", { enumerable: true, get: function () { return utils_1.extractMessage; } });
var handling_1 = require("./handling");
Object.defineProperty(exports, "handleApiError", { enumerable: true, get: function () { return handling_1.handleApiError; } });
var upsert_1 = require("./upsert");
Object.defineProperty(exports, "PineconeBatchUpsertError", { enumerable: true, get: function () { return upsert_1.PineconeBatchUpsertError; } });
